tests/integration/plots/test_json.py

```diff
@@ -0,0 +1,246 @@
+    )
+    split_json_content = deepcopy(
+        dpath.util.get(split_json_result, [filename, 0, "content"])
+    )
+    dpath.util.set(json_content, ["data"], {})
+    dpath.util.set(split_json_content, ["data"], {})
+    assert json_content == split_json_content
+
+
+def verify_vega_props(filename, json_result, title, x, y, **kwargs):
+    data = json_result[filename]
+    assert len(data) == 1
+    data = first(data)
+
+    assert dpath.util.get(data, ["content", "title"]) == title
+
+    assert (
+        dpath.util.get(data, ["content", "spec", "encoding", "x", "field"])
+        == x
+    )
+    assert (
+        dpath.util.get(data, ["content", "spec", "encoding", "y", "field"])
+        == y
+    )
+
+
+@pytest.mark.vscode
+def test_no_plots(tmp_dir, scm, dvc, capsys):
+    html_path, json_result, split_json_result = call(capsys)
+    assert os.path.exists(html_path)
+    assert json_result == {}
+    assert split_json_result == {}
+
+
+@pytest.mark.vscode
+def test_repo_with_plots(
+    tmp_dir, scm, dvc, capsys, run_copy_metrics, repo_with_plots
+):
+    repo_state = repo_with_plots()
+
+    image_v1, linear_v1, confusion_v1, confusion_props = next(repo_state)
+
+    html_path, json_result, split_json_result = call(capsys)
+
+    verify_image(
+        tmp_dir, "workspace", "image.png", image_v1, html_path, json_result
+    )
+
+    verify_vega(
+        "workspace",
+        "linear.json",
+        ["x", "y"],
+        linear_v1,
+        html_path,
+        json_result,
+        split_json_result,
+    )
+
+    verify_vega(
+        "workspace",
+        "confusion.json",
+        ["actual", "predicted"],
+        confusion_v1,
+        html_path,
+        json_result,
+        split_json_result,
+    )
+    verify_vega_props("confusion.json", json_result, **confusion_props)
+
+    image_v2, linear_v2, confusion_v2, confusion_props = next(repo_state)
+
+    html_path, json_result, split_json_result = call(capsys, subcommand="diff")
+
+    verify_image(
+        tmp_dir, "workspace", "image.png", image_v2, html_path, json_result
+    )
+    verify_image(
+        tmp_dir, "HEAD", "image.png", image_v1, html_path, json_result
+    )
+
+    verify_vega(
+        ["HEAD", "workspace"],
+        "linear.json",
+        ["x", "y"],
+        linear_v2 + linear_v1,
+        html_path,
+        json_result,
+        split_json_result,
+    )
+
+    verify_vega(
+        ["HEAD", "workspace"],
+        "confusion.json",
+        ["actual", "predicted"],
+        confusion_v2 + confusion_v1,
+        html_path,
+        json_result,
+        split_json_result,
+    )
+    verify_vega_props("confusion.json", json_result, **confusion_props)
+    from dvc.utils.fs import remove
+
+    # even if there is no data, call should be successful
+    remove(tmp_dir / ".dvc" / "cache")
+    remove("linear.json")
+    remove("confusion.json")
+    remove("image.png")
+    call(capsys)
+    call(capsys, subcommand="diff")
```

Comment:
Isn't this covered by `test_no_plots`? See suggestion above
```suggestion
```

Reply:
Not quite, in case of no plots at all, we get empty results when collecting plots definitions. 
In this case, we have definitions but we don't have the data. That will make a difference down the line when generating visualizations. This part can detect the problem with no `final_props` in `convert.py` while `no-plots` cannot because it never reaches the converter.